File: Bulgarian Competitions/BNOI - Group A and B/Sum Of Primes/Sum Of Primes.cs
using System;
using System.Collections.Generic;

class SumOfPrimes
{
    static void Main()
    {
        int sum = int.Parse(Console.ReadLine());
        List<int> primes = new List<int>();
        for (int i = 2; i < sum; i++)
        {
            bool isPrime = true;
            for (int j = 2; j < i; j++)
            {
                if (i % j == 0)
                {
                    isPrime = false;
                    break;
                }
            }

            if (isPrime) primes.Add(i);
        }

        int count = 0;
        for (int i = 0; i < primes.Count; i++)
        {
            for (int j = 0; j < primes.Count; j++)
            {
                if (primes[i] + primes[j] == sum) count++;
            }
        }

        Console.WriteLine(count / 2 + (count % 2 == 0 ? 0 : 1));
    }
}

File: Bulgarian Competitions/BNOI - Group A and B/Sum of Primes/Sum of Primes.js
function solve(args) {
    var sum = parseInt(args[0]);
    var primes = [];
    for (var i = 2; i < sum; i++) {
        var isPrime = true;
        for (var j = 2; j < i; j++) {
            if (i % j == 0) {
                isPrime = false;
                break;
            }
        }

        if (isPrime) primes.push(i);
    }

    var count = 0;
    for (var i = 0; i < primes.length; i++) {
        for (var j = 0; j < primes.length; j++) {
            if (primes[i] + primes[j] == sum) count++;
        }
    }

    return Math.floor(count / 2) + (count % 2 == 0 ? 0 : 1);
}